tools/query/src/query_type.rs

```diff
@@ -96,84 +103,264 @@ pub enum QueryType {
-
-          Ok(json!({
-            "address": addr
-          }))
-        },
-        _ => { bail!("Not implemented for type: {:?}", self) }
-        // QueryType::BlockHeight => todo!(),
-        // QueryType::Resources { account } => todo!(),
-        // QueryType::MoveValue { account, module_name, struct_name, key_name } => todo!(),
+            QueryType::Balance { account } => {
+                let json_data = get_account_balance_libra(&client, *account).await?;
+                Ok(OutputType::Json(serde_json::to_string_pretty(&json_data)?))
+            }
+            QueryType::Tower { account } => {
+                let tower_state = get_tower_state(&client, *account).await?;
+                let json_data = serde_json::to_value(tower_state)?;
+                Ok(OutputType::Json(serde_json::to_string_pretty(&json_data)?))
+            }
+            QueryType::View {
+                function_id,
+                type_args,
+                args,
+            } => {
+                let json_data =
+                    fetch_and_display(&client, function_id, type_args.to_owned(), args.to_owned())
+                        .await?;
+                Ok(OutputType::Json(serde_json::to_string_pretty(&json_data)?))
+            }
+            QueryType::Epoch => {
+                let json_data = fetch_and_display(
+                    &client,
+                    "0x1::reconfiguration::get_current_epoch",
+                    None,
+                    None,
+                )
+                .await?;
+                Ok(OutputType::Json(serde_json::to_string_pretty(&json_data)?))
+            }
+            QueryType::BlockHeight => {
+                let json_data =
+                    fetch_and_display(&client, "0x1::block::get_current_block_height", None, None)
+                        .await?;
+                Ok(OutputType::Json(serde_json::to_string_pretty(&json_data)?))
+            }
+            QueryType::Resources { account } => {
+                let res = client
+                    .get_account_resources(*account)
+                    .await?
+                    .into_inner()
+                    .into_iter()
+                    .map(|resource| {
+                        let mut map = serde_json::Map::new();
+                        map.insert(resource.resource_type.to_string(), resource.data);
+                        serde_json::Value::Object(map)
+                    })
+                    .collect::<Vec<serde_json::Value>>();
+                Ok(OutputType::Json(serde_json::to_string_pretty(&res)?))
+            }
+            QueryType::LookupAddress { auth_key } => {
+                let addr = lookup_originating_address(&client, auth_key.to_owned()).await?;
 
-    }
+                Ok(OutputType::Json(serde_json::to_string(&json!({
+                    "address": addr
+                }))?))
+            }
+            QueryType::MoveValue {
```

Comment:
I would first move this to it's own file.
Also I'm sure BCS can do a lot of this conversion.  Look for examples in Diem of `let x: SomeMoveValue = BCS:from_bcs(resource);